artichoke-backend/src/extn/core/integer/div.rs

```diff
@@ -0,0 +1,109 @@
+use artichoke_core::value::Value as _;
+
+use crate::convert::Convert;
+use crate::extn::core::exception::{Fatal, RubyException, TypeError, ZeroDivisionError};
+use crate::extn::core::float::Float;
+use crate::types::{self, Int};
+use crate::value::Value;
+use crate::Artichoke;
+
+pub fn method(
+    interp: &Artichoke,
+    value: Value,
+    other: Value,
+) -> Result<Value, Box<dyn RubyException>> {
+    let x = value.try_into::<Int>().map_err(|_| {
+        Fatal::new(
+            interp,
+            "Unable to extract Rust Integer from Ruby Integer receiver",
+        )
+    })?;
+    let pretty_name = other.pretty_name();
+    if let Ok(y) = other.clone().try_into::<Int>() {
+        if y == 0 {
+            Err(Box::new(ZeroDivisionError::new(interp, "divided by 0")))
+        } else {
+            Ok(interp.convert(x / y))
+        }
+    } else if let Ok(y) = other.try_into::<types::Float>() {
+        if y == 0.0 {
+            if x > 0 {
+                Ok(interp.convert(Float::INFINITY))
+            } else if x < 0 {
+                Ok(interp.convert(Float::NEG_INFINITY))
+            } else {
+                // x == 0
+                Ok(interp.convert(Float::NAN))
+            }
```

Comment:
not sure if this is necessary or if Rust float division does the right thing here.